msteams-platform/apps-in-teams-meetings/build-apps-for-anonymous-user.md

```diff
@@ -59,76 +125,91 @@ The [get members](/microsoftteams/platform/bots/how-to/get-teams-context#fetch-t
   "name": "<AnonTest (Guest)>",  
   "tenantId": "<GUID2>", 
   "userRole": "anonymous" 
-} 
+}
 ```
 
 | **Property name** | **Description** |
 | --- | --- |
 | `id` | Unique generated value for the anonymous user. |
 | `name` | Name provided by the anonymous user when joining the meeting. |
+| `userRole` | `Anonymous` represents anonymous user. |
 | `tenantId` | Tenant ID of the meeting organizer. |
 | `userRole` | `anonymous`, represents anonymous user. |
 
-> [!NOTE]
-> The ID received from the bot APIs and the Microsoft Teams JavaScript client library (TeamsJS) API are not the same.
-
 ### ConversationUpdate activity MembersAdded and MembersRemoved
 
+`MembersAdded`
+
 ```csharp
-{ 
-  "membersAdded": [ 
-    { 
-      "id": "<GUID1>" 
-    } 
-  ], 
-  "type": "conversationUpdate", 
-  "timestamp": "<timestamp>", 
-  "id": "<event unique identifier>", 
-  "channelId": "msteams", 
-  "serviceUrl": "<serviceURL>", 
-  "from": { 
-    "id": "<GUID2>" 
-  }, 
-  "conversation": { 
-    "isGroup": true, 
-    "tenantId": "<tenant id>", 
-    "id": "<conversation id>" 
-  }, 
-  "recipient": { 
-    "id": "<bot id>", 
-    "name": "<bot name>" 
-  }, 
-  "channelData": { 
-    "tenant": { 
-      "id": "<tenant id>" 
-    }, 
-    "source": null, 
-    "meeting": { 
-      "id": "<meeting id>" 
-    } 
-  } 
-} 
+protected override async Task OnTeamsMembersAddedAsync(IList<TeamsChannelAccount> membersAdded, TeamInfo teamInfo, ITurnContext<IConversationUpdateActivity> turnContext, CancellationToken cancellationToken)
+  {
+     foreach (var teamMember in membersAdded)
+     {
+         if (teamMember.UserRole == "anonymous" )
```

Comment:
Add comment before if like:

// If UserRole == "anonymous", it indicates an anonymous user

Reply:
updated